marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/graph/searchNodes.sjs

```diff
@@ -68,11 +72,38 @@ fn.collection(entityLib.getModelCollection()).toArray().forEach(model => {
 
   if(selectedFacets && selectedFacets.relatedEntityTypeIds && selectedFacets.relatedEntityTypeIds.includes(entityName)) {
     allEntityTypeIRIs.push(sem.iri(entityNameIri));
+
+    let entityProperties = model.definitions[entityName].properties;
+
+    for(let entityPropertyName in entityProperties){
+      let entityPropertyValue = entityProperties[entityPropertyName];
+
+
+      if(entityPropertyValue["relatedEntityType"] != null){
+        predicateList.push(sem.iri(entityNameIri+"/"+entityPropertyName));
+
+      }else{
+        if(entityPropertyValue["items"] != null){
+          let items = entityPropertyValue["items"]
+          if(items["relatedEntityType"] != null){
+            predicateList.push(sem.iri(entityNameIri+"/"+entityPropertyName));
+
+          }
+
+        }
+      }
+    }
+    if(predicateList.length >= 1){
+      hashmapPredicate.set(entityName, predicateList);
```

Comment:
You are never reseting predicateList. On the second and next loops it would contain the predicates from previous entities.